server/re/src/mailMS.cpp

```diff
@@ -87,20 +88,11 @@ int msiSendMail( msParam_t* xtoAddr, msParam_t* xsubjectLine, msParam_t* xbody,
         return SYS_MALLOC_ERR;
     }
 
-#ifdef solaris_platform
-    sprintf( mailStr, "cat %s| mail  '%s'", fName, toAddr );
-#else /* tested for linux - not sure how other platforms operate for subject */
-    if ( strlen( subjectLine ) > 0 ) {
-        sprintf( mailStr, "cat %s| mail -s '%s'  '%s'", fName, subjectLine, toAddr );
-    }
-    else {
-        sprintf( mailStr, "cat %s| mail  '%s'", fName, toAddr );
-    }
-#endif
     int ret = 0;
-    ret = system( mailStr );
+    namespace bp = boost::process;
+    ret = bp::system(bp::search_path("mail"), "-s", subjectLine, toAddr, bp::std_in < fName);
     if ( ret ) {
-        irods::log( ERROR( ret, "mailStr command returned non-zero status" ) );
+        irods::log(ERROR(ret, "boost::process::system command returned non-zero status"));
```

Comment:
Boost process is an implementation detail and doesn't provide enough information to help the admin know where to start looking.

How do you feel about something like ... `msiSendMail error: <error_code>`?

That immediately tells the admin it's specific to this MSI.